### PR TITLE
【映射文档】补全缺失的映射文档 part.2

### PR DESCRIPTION
**Pull request by RedContritio**
- 添加 21 篇映射文档
- 仍然缺失的映射文档有 

    > torch.Tensor.max
torch.Tensor.min
torch.Tensor.rename

    其中 `torch.Tensor.min` 和 `torch.Tensor.max` 逻辑仍在检查，`torch.Tensor.rename` 疑似不支持。

### CODE REVIEW

**Comment by zhwesky2010** [comment]
`torch.Tensor.rename` 目前不支持，无需编写

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_empty.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+             name=None)
+```
+
+PyTorch 相比 Paddle 支持更多其他参数，具体如下：
+
+### 参数映射
+
+| PyTorch       | PaddlePaddle | 备注                                                         |
+| ------------- | ------------ | ------------------------------------------------------------ |
+| *size         | shape        | 表示输出形状大小， PyTorch 是多个元素， Paddle 是列表或元组，需要转写。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_empty.md]: Pytorch是可变参数用法

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_full.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+PyTorch 相比 Paddle 支持更多其他参数，具体如下：
+
+### 参数映射
+
+| PyTorch       | PaddlePaddle | 备注                                                         |
+| ------------ | ----------- | ----------------------------------------------------------- |
+| size          | shape        | 表示创建 Tensor 的形状，仅参数名不一致。                     |
+| fill_value    | fill_value   | 表示初始化输出 Tensor 的常量数据的值                         |
+| dtype         | dtype        | 表示输出 Tensor 类型。                                       |
+| layout        | -            | 表示布局方式，Paddle 无此参数，一般对网络训练结果影响不大，可直接删除。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_full.md]: 按torch参数的顺序写吧

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_ones.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+             name=None)
+```
+
+PyTorch 相比 Paddle 支持更多其他参数，具体如下：
+
+### 参数映射
+
+| PyTorch       | PaddlePaddle | 备注                                                         |
+| ------------- | ------------ | ------------------------------------------------------------ |
+| *size         | shape        | 表示输出形状大小， PyTorch 是多个元素， Paddle 是列表或元组，需要转写。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_ones.md]: 可变参数用法

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_zeros.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+             name=None)
+```
+
+PyTorch 相比 Paddle 支持更多其他参数，具体如下：
+
+### 参数映射
+
+| PyTorch       | PaddlePaddle | 备注                                                         |
+| ------------- | ------------ | ------------------------------------------------------------ |
+| *size         | shape        | 表示输出形状大小， PyTorch 是多个元素， Paddle 是列表或元组，需要转写。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_zeros.md]: 多个参数 都改成 **可变参数用法** 吧

File: docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.BoolTensor.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+Paddle 比 PyTorch 支持更多参数，具体如下：
+
+### 参数映射
+
+| PyTorch | PaddlePaddle | 备注                                                        |
+| ------- | ------------ | ----------------------------------------------------------- |
+| data    | data         | 要转换的数据。 |
+| -       | dtype        | Tensor 的数据类型，PyTorch 无此参数，Paddle 需设置为 'bool'。   |
+| -       | place        | Tensor 的设备，PyTorch 无此参数，Paddle 需设置为 'cuda' 。         |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.BoolTensor.md]: 这个是设置为 'cuda' 还是 'gpu'

File: docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.ByteTensor.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+Paddle 比 PyTorch 支持更多参数，具体如下：
+
+### 参数映射
+
+| PyTorch | PaddlePaddle | 备注                                                        |
+| ------- | ------------ | ----------------------------------------------------------- |
+| data    | data         | 要转换的数据。 |
+| -       | dtype        | Tensor 的数据类型，PyTorch 无此参数，Paddle 需设置为 'uint8'。   |
+| -       | place        | Tensor 的设备，PyTorch 无此参数，Paddle 需设置为 'cuda' 。         |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.ByteTensor.md]: 这个是设置为 'cuda' 还是 'gpu'

File: docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.FloatTensor.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+Paddle 比 PyTorch 支持更多参数，具体如下：
+
+### 参数映射
+
+| PyTorch | PaddlePaddle | 备注                                                        |
+| ------- | ------------ | ----------------------------------------------------------- |
+| data    | data         | 要转换的数据。 |
+| -       | dtype        | Tensor 的数据类型，PyTorch 无此参数，Paddle 需设置为 'float32'。   |
+| -       | place        | Tensor 的设备，PyTorch 无此参数，Paddle 需设置为 'cuda' 。         |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.FloatTensor.md]: 这个是设置为 'cuda' 还是 'gpu'

File: docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.IntTensor.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+Paddle 比 PyTorch 支持更多参数，具体如下：
+
+### 参数映射
+
+| PyTorch | PaddlePaddle | 备注                                                        |
+| ------- | ------------ | ----------------------------------------------------------- |
+| data    | data         | 要转换的数据。 |
+| -       | dtype        | Tensor 的数据类型，PyTorch 无此参数，Paddle 需设置为 'int32'。   |
+| -       | place        | Tensor 的设备，PyTorch 无此参数，Paddle 需设置为 'cuda' 。         |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.IntTensor.md]: 这个是设置为 'cuda' 还是 'gpu'

File: docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.LongTensor.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+Paddle 比 PyTorch 支持更多参数，具体如下：
+
+### 参数映射
+
+| PyTorch | PaddlePaddle | 备注                                                        |
+| ------- | ------------ | ----------------------------------------------------------- |
+| data    | data         | 要转换的数据。 |
+| -       | dtype        | Tensor 的数据类型，PyTorch 无此参数，Paddle 需设置为 'int64'。   |
+| -       | place        | Tensor 的设备，PyTorch 无此参数，Paddle 需设置为 'cuda' 。         |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/cuda/torch.cuda.LongTensor.md]: 这个是设置为 'cuda' 还是 'gpu'

File: docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.rpc.init_rpc.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+### 参数映射
+
+| PyTorch             | PaddlePaddle        | 备注 |
+| ------------------- | ------------------- | -- |
+| name                | name                | worker 名字。 |
+| backend             | -                   | RPC 后台实现类型，Paddle 无此参数，暂无转写方式。 |
+| rank                | rank                | worker 的 ID，仅参数默认值不同。 |
+| world_size          | world_size          | workers 的数量。 |
+| rpc_backend_options | -                   | 传递给 worker 创建时的配置项，Paddle 无此参数，暂无转写方式。 |
+| -                   | master_endpoint     | master 的 IP 地址。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/distributed/torch.distributed.rpc.init_rpc.md]: 仅paddle参数更多：需要写明paddle怎么设置

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Generator.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+paddle.framework.core.default_cpu_generator()
+```
+
+```python
+# PyTorch 写法
+torch.Generator(device="cuda")
+
+# Paddle 写法
+device = paddle.device.get_device()
+paddle.framework.core.default_cuda_generator(int(device[-1]))
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Generator.md:27]: 直接把id设为0，看起来简单点

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Size.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+Paddle 无此 API，需要组合实现。
+
+### 转写示例
+
+```python
+# PyTorch 写法
+torch.Size([1])
+
+# Paddle 写法
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Size.md:17]: Paddle 写法，通过tuple代替

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Size.md [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+tuple([1])
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Size.md]: (1, )

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Size.md [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+```
+
+```python
+# PyTorch 写法
+torch.Size([2, 3])
+
+# Paddle 写法
+tuple([2, 3])
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Size.md]: (2, 3)

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.device.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+Paddle 无此 API，需要组合实现。
+
+### 转写示例
+
+```python
+# PyTorch 写法
+torch.device('cuda', 0)
+
+# Paddle 写法
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.device.md:17]: Paddle 写法，通过字符串代替

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.inference_mode.md [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+Paddle 无此 API，需要组合实现。
+
+### 转写示例
+
+```python
+# PyTorch 写法
+@torch.inference_mode(False)
+
+# Paddle 写法
+@empty_decorator()
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.inference_mode.md]: 这个就比较迷惑了，写清楚实际功能与含义吧

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.index_copy_.md [hunk 17]
```diff
@@ -0,0 +1,29 @@
+## [ 组合替代实现 ]torch.Tensor.index_copy_
+
+### [torch.Tensor.index_copy_](https://pytorch.org/docs/stable/generated/torch.Tensor.index_copy_.html)
+
+```python
+torch.Tensor.index_copy_(dim, index, source)
+```
+
+Paddle 无此 API，需要组合实现。
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.index_copy_.md:9]: https://pytorch.org/docs/stable/generated/torch.index_copy.html#torch-index-copy 

看下index_add能否满足，原来的组合写法可能也不是最好的，可以进行matcher升级

**Comment by zhwesky2010** [comment]
@RedContritio 需要修改一下comment

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Generator.md (hunk 11 above)
Comment (RedContritio) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.Generator.md:27]: 反馈，这里的 device 实际上是字符串 `'gpu:0'`，是取 id 作为 int 参数传入。

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.index_copy_.md (hunk 17 above)
Comment (RedContritio) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.index_copy_.md:9]: 目前还没法直接替换，因为 `index_add` 是 `tensor += alpha * other`，但是 `index_copy` 是 `tensor[...] = other[...]`，目前只能通过组合来实现。

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_full.md [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+| fill_value    | fill_value   | 表示初始化输出 Tensor 的常量数据的值                         |
+| dtype         | dtype        | 表示输出 Tensor 类型。                                       |
+| device        | -            | 表示 Tensor 存放设备位置，Paddle 无此参数，需要转写。    |
+| requires_grad | -            | 表示是否计算梯度，Paddle 无此参数，需要转写。            |
+| layout        | -            | 表示布局方式，Paddle 无此参数，一般对网络训练结果影响不大，可直接删除。 |
+| pin_memory    | -            | 表示是否使用锁页内存， Paddle 无此参数，需要转写。       |
+
+### 转写示例
+
+#### out：指定输出
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_full.md]: 没有out，多的是pin_memory

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.modules.utils._pair.md [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+torch.nn.modules.utils._pair(x)
+```
+
+Paddle 无此 API，需要组合实现。
+
+### 转写示例
+
+```python
+# PyTorch 写法
+torch.nn.modules.utils._pair(x)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.modules.utils._pair.md:15]: torch.nn.modules.utils._pair(2)

File: docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.inference_mode.md (hunk 16 above)
Comment (zhwesky2010) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/ops/torch.inference_mode.md]: 这个就写直接删去吧。
自动转写时：因为无法实现删除的效果，所以才弄了一个空装饰器；
人工转写时，告诉他思路是**直接删除**就可以，不用机械的照搬AST代码里的实现

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.modules.utils._pair.md (hunk 19 above)
Comment (RedContritio) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.modules.utils._pair.md:15]: 这里是接受的任意参数 x

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_full.md [hunk 20]
```diff
@@ -31,34 +31,34 @@ PyTorch 相比 Paddle 支持更多其他参数，具体如下：
 
 ### 转写示例
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_full.md:32]: 这种 `torch.Tensor.new_*` 的API，需要说明下dtype的差异，这些地方都需要转写：`dtype=x.dtype`

Comment (zhwesky2010) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_full.md:32]: 其他的有很多这种API，都需要一并处理下，并且检查下转换工具里，处理了这个差异没有

Comment (RedContritio) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_full.md:32]: 转换工具处理了，已更新文档对应部分

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_empty.md [hunk 21]
```diff
@@ -21,7 +21,7 @@ PyTorch 相比 Paddle 支持更多其他参数，具体如下：
 | PyTorch       | PaddlePaddle | 备注                                                         |
 | ------------- | ------------ | ------------------------------------------------------------ |
 | *size         | shape        | 表示输出形状大小， PyTorch 是可变参数用法， Paddle 是列表或元组，需要转写。 |
-| dtype         | dtype        | 表示输出 Tensor 类型。                                       |
+| dtype         | dtype        | 表示输出 Tensor 类型，如果没有指定，默认使用当前对象的 dtype，需要转写。    |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.new_empty.md:24]: 这个最好分开写，比如torch的表现，然后再写paddle的表现以及应该怎么去设置。

毕竟每一栏里都是torch、paddle两个参数，混着写容易不太清晰，后面PR修改吧

**Review by zhwesky2010 — APPROVED** [review]
